tools/generate_code_cache.js

```diff
@@ -52,29 +52,26 @@ function formatSize(num) {
  * initializers of the code cache.
  *
  * @param {string} key ID of the builtin module
- * @param {Buffer} cache Code cache of the builtin module
+ * @param {Uint8Array} cache Code cache of the builtin module
  * @return { definition: string, initializer: string }
  */
 function getInitalizer(key, cache) {
-  const defName = key.replace(/\//g, '_').replace(/-/g, '_');
-  const definition = `static uint8_t ${defName}_raw[] = {\n` +
+  const defName = `${key.replace(/\//g, '_').replace(/-/g, '_')}_raw`;
+  const definition = `static const uint8_t ${defName}[] = {\n` +
                      `${cache.join(',')}\n};`;
   const source = getSource(key);
   const sourceHash = hash(source);
-  const initializer = `
-  v8::Local<v8::ArrayBuffer> ${defName}_ab =
-    v8::ArrayBuffer::New(isolate, ${defName}_raw, ${cache.length});
-  v8::Local<v8::Uint8Array> ${defName}_array =
-    v8::Uint8Array::New(${defName}_ab, 0, ${cache.length});
-  target->Set(context,
-              FIXED_ONE_BYTE_STRING(isolate, "${key}"),
-              ${defName}_array).FromJust();
-  `;
-  const hashIntializer = `
-  target->Set(context,
-              FIXED_ONE_BYTE_STRING(isolate, "${key}"),
-              OneByteString(isolate, "${sourceHash}")).FromJust();
-  `;
+  const bytes = `UnionBytes(${defName}, arraysize(${defName}))`;
+  const initializer =
+    'code_cache_.insert(std::make_pair(\n' +
```

Comment:
Doesn't ``` `code_cache_.emplace("${key}",  ${bytes})` ``` work?
Or even ``` `code_cache_.emplace("${key}",  {${defName}, arraysize(${defName})})` ```

Reply:
The latter doesn't seem to work without `std::initializer_list`, simply using the constructor is a bit more readable. I'll just use emplace, thanks

Reply:
Less cruft while still readable - win win ;)